### PR TITLE
feat(updater): show download stats below progress bar (#159)

### LINKED ISSUE #159: feat(updater): display progress and speed below progress bar
# Overview
While the updater is downloading an update, we want to display the current download progress and speed below the progress bar. This helps users understand how much of the download is complete and the current download rate.

**Planned UI Feedback:**  
`45 MB / 120 MB (1.2 MB/s)`  

## Steps
1. Update `useUpdater.ts` to track the already downloaded data and approximate download speed.  
2. Format the download size and speed to display in MB and MB/s. (→ Create a function with JSDoc in the `renderer/lib/updater` folder)  
3. Add tests for your created format function.
4. **(Optional)** Feel free to add yourself to the Contributors List in the README.

## Current UI
Image

### PR DESCRIPTION
**Pull request by BenedictHomuth**
## Description
This PR adds download progress and speed information below the progress bar during application updates. It displays the amount of data transferred, the total size, and the download speed (e.g., `45 MB / 120 MB (1.2 MB/s)`).

 Implements/Resolves #159

 ## Motivation and Context
 Previously, users only saw a percentage progress bar when downloading an update. Adding concrete transfer statistics gives better feedback, especially on slower connections, so users know exactly how
      much data is being downloaded and at what speed.

*Note: The download speed and progress are calculated entirely locally using the `electron-updater` package, without relying on any third-party services for bandwidth monitoring.*

 ## Changes included
 - Created a `formatBytes` utility in `renderer/lib/updater/formatBytes.ts` to convert raw byte counts into human-readable strings (KB, MB, GB, etc.).
 - Added unit tests for the formatter in `renderer/tests/lib/updater/formatBytes.test.ts`.
 - Updated `UpdaterContext.tsx` and `useUpdater.ts` to capture and expose the full `ProgressInfo` payload from `electron-updater`.
 - Passed the `progressInfo` state through `update.tsx` to the `UpdaterProvider`.
 - Modified `UpdaterProgressBar.tsx` to display the formatted download statistics directly below the progress bar.

 ## Maintenance & CI Fixes (Unrelated to Feature)
 To ensure the build and linting pipelines passed successfully for this PR, I addressed the following pre-existing issues found during validation:
 - **Linting:** Removed an unused `isFormValid` variable in `renderer/pages/[locale]/profile/create/index.tsx` that was causing ESLint failures.
 - **TypeScript:** Updated `renderer/tests/hooks/useDartGame.test.ts` to include the missing `currentLegStartingPlayerIndex` property in mock states, resolving strict type-checking errors.
 - **Formatting:** Ran Prettier across the modified files to ensure strict adherence to the project's styling rules.

 ## How Has This Been Tested?
 - Added `vitest` unit tests to ensure `formatBytes` works correctly for different sizes and edge cases (like `0 Bytes`).
 - Verified that the `ProgressInfo` from `electron-updater` successfully populates the UI with live-updating statistics.
 - Ran the full suite of 209 tests, linting, and TypeScript checks to confirm 100% pass rate.

### CODE REVIEW

File: renderer/lib/updater/formatBytes.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Formats a given number of bytes into a human-readable string.
+ *
+ * @param bytes - The number of bytes to format.
+ * @param decimals - The number of decimal places to include (default: 2).
+ * @returns A formatted string with the appropriate unit (e.g., "1.50 MB").
+ */
+export const formatBytes = (bytes: number, decimals = 2): string => {
+  if (!+bytes) return "0 Bytes";
```
Comment (timderes) [inline renderer/lib/updater/formatBytes.ts]: Please tell me this is AI. I don't mind keeping it, but i never saw this before 😆 

`!Number(bytes)` is maybe better to read

File: renderer/lib/updater/formatBytes.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  const k = 1024;
+  const dm = decimals < 0 ? 0 : decimals;
+  const sizes = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
+
+  const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+  return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`;
```
Comment (timderes) [inline renderer/lib/updater/formatBytes.ts]: Maybe we can use `toLocaleString` here so the formatting matches the user's locale (e.g. `,` in Germany).

File: renderer/components/updater/UpdaterProgressBar.tsx [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+  const { progress, progressInfo, status } = useUpdater();
 
   if (status === "downloading") {
-    return <Progress {...props} w={props.w ?? 200} value={progress ?? 0} />;
+    return (
+      <>
+        <Progress {...props} w={props.w ?? 200} value={progress ?? 0} />
+        {progressInfo && (
+          <Text size="xs" ta="center" mt="xs">
+            {formatBytes(progressInfo.transferred)} /{" "}
```
Comment (timderes) [inline renderer/components/updater/UpdaterProgressBar.tsx]: Reference to the comment in `formatBytes.ts`:

Mantine provides a NumberFormatter component [https://mantine.dev/core/number-formatter/](https://mantine.dev/core/number-formatter/)

Maybe we can use it at the UI level to handle locale formatting, while keeping the helper function as it is

**Review by timderes — COMMENTED** [review]
A few minor suggestions from my side. Open to feedback :) @BenedictHomuth

**Comment by BenedictHomuth** [comment]
All your comments are addressed hopefully. Give it a new look! :)

**Review by timderes — APPROVED** [review]
LGTM! Added a missing space between value and unit. Ready to merge.
